### PR TITLE
D3ASIM-3631: Added time_slot parameter to order and trade objects and…

### PR DESCRIPTION
**Pull request by hannesdiedrich**
… future_market_duration to GlobalConfig.

### CODE REVIEW

File: d3a_interface/data_classes.py [hunk 1]
```diff
@@ -150,6 +151,7 @@ def from_dict(offer: Dict) -> "Offer":
             requirements=offer.get("requirements"))
 
     def __eq__(self, other) -> bool:
+        # TODO: maybe add the self.time_slot here, but may make the comparing slower and not needed
```
Comment (spyrostz) [inline d3a_interface/data_classes.py:154]: Datetime comparisons are integer comparisons IIRC (essentially only the Unix timestamp is compared IIRC) therefore I would add it too

File: d3a_interface/data_classes.py [hunk 2]
```diff
@@ -270,17 +273,18 @@ def from_json(trade_bid_offer_info: str) -> "TradeBidOfferInfo":
...
                  offer_bid_trade_info: Optional[TradeBidOfferInfo] = None,
                  seller_origin: Optional[str] = None, buyer_origin: Optional[str] = None,
                  fee_price: Optional[float] = None, seller_origin_id: Optional[str] = None,
                  buyer_origin_id: Optional[str] = None, seller_id: Optional[str] = None,
-                 buyer_id: Optional[str] = None):
+                 buyer_id: Optional[str] = None, time_slot: DateTime = None):
 
         self.id = str(id)
-        self.time = time
+        self.time = time  # Actual time if creation
```
Comment (Hassan754) [inline d3a_interface/data_classes.py:286]:
```suggestion
        self.time = time  # Actual time of creation
```

File: d3a_interface/data_classes.py [hunk 3]
```diff
@@ -270,17 +273,18 @@ def from_json(trade_bid_offer_info: str) -> "TradeBidOfferInfo":
...
-    def __init__(self, id: str, time: datetime, offer_bid: Union[Offer, Bid],
+    def __init__(self, id: str, time: DateTime, offer_bid: Union[Offer, Bid],
                  seller: str, buyer: str, residual: Optional[Union[Offer, Bid]] = None,
                  already_tracked: bool = False,
                  offer_bid_trade_info: Optional[TradeBidOfferInfo] = None,
                  seller_origin: Optional[str] = None, buyer_origin: Optional[str] = None,
                  fee_price: Optional[float] = None, seller_origin_id: Optional[str] = None,
                  buyer_origin_id: Optional[str] = None, seller_id: Optional[str] = None,
-                 buyer_id: Optional[str] = None):
+                 buyer_id: Optional[str] = None, time_slot: DateTime = None):
```
Comment (Hassan754) [inline d3a_interface/data_classes.py:283]:
```suggestion
                 buyer_id: Optional[str] = None, time_slot: Optional[DateTime] = None):
```

File: d3a_interface/data_classes.py [hunk 4]
```diff
@@ -37,10 +36,11 @@ def json_datetime_serializer(date_obj: DateTime):
 class BaseBidOffer:
     """Base class defining shared functionality of Bid and Offer market structures."""
     def __init__(self, id: str, time: DateTime, price: float, energy: float,
-                 original_price: Optional[float] = None,
+                 original_price: Optional[float] = None, time_slot: DateTime = None,
                  attributes: Dict = None, requirements: List[Dict] = None):
         self.id = str(id)
-        self.time = time
+        self.time = time  # Actual time if creation
```
Comment (Hassan754) [inline d3a_interface/data_classes.py:42]:
```suggestion
        self.time = time  # Actual time of creation
```

**Pull request merged**